### PR TITLE
block login flows when inside the cloud console

### PR DESCRIPTION
**Pull request by yugangw-msft**
//cc: @jluk @yangl900 

- [na ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [na] Each command and parameter has a meaningful description.
- [na] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by jluk** [comment]
@yugangw-msft can this wait until Mon/Tues? I'd like to make sure there are no tooling scenarios dependent on service principals.

As an aside, the product name is "Cloud Shell" not "Console" please update the error string.

Suggestion:
"Executing the 'login' command inside Azure Cloud Shell is unnecessary as your accounts have already been configured."

File: src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py [hunk 1]
```diff
@@ -91,6 +91,14 @@ def login(username=None, password=None, service_principal=None, tenant=None,
 
     profile = Profile()
 
+    if os.environ.get('ACC_CLOUD', None):
+        console_tokens = os.environ.get('AZURE_CONSOLE_TOKENS', None)
+        if console_tokens:
+            return profile.find_subscriptions_in_cloud_console(re.split(';|,', console_tokens))
+        else:
+            raise CLIError("Executing the 'login' command inside the Azure Cloud Console is unnecessary as"
```
Comment (jluk) [inline src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py]: Update error string:

"Azure Cloud Shell relies on the user account it was initially launched under, as a result az login is disabled."

**Comment by yugangw-msft** [comment]
> @yugangw-msft can this wait until Mon/Tues? I'd like to make sure there are no tooling scenarios dependent on service principals.

Yes, I can wait

**Comment by tjprescott** [comment]
This is more a philosophical question, but rather than raise an error that "This command doesn't work", would it be more logical to suppress the login/logout commands entirely (the way az interactive has to do with the `interactive` command)?

**Comment by jluk** [comment]
@tjprescott My concern there is leaving out the cmd will make it seem like a bug without a way to discover it was meant to be left out. We'd get support tickets opened in that scenario, but with an explicit error message the user at least knows it is intended behavior.

**Comment by yugangw-msft** [comment]
The other main reason is the Cloud Shell needs to invoke `az login` command with raw tokens to establish the accounts, so we can't really block the whole command, rather just block a few supported flows.

**Comment by yugangw-msft** [comment]
@jluk, I applied the same error message on `az logout`.  Can I merge the change now?

**Comment by jluk** [comment]
@yugangw-msft thanks I think we are good to go

**Pull request merged**